app/src/main/java/com/ml/shubham0204/docqa/ui/screens/docs/DocsViewModel.kt

```diff
@@ -55,6 +59,38 @@ class DocsViewModel(
         }
     }
 
+    suspend fun addDocumentFromUrl(url: String, context: Context, callback: (Boolean) -> Unit) = withContext(Dispatchers.IO){
+        try {
+            val connection = URL(url).openConnection() as HttpURLConnection
+            connection.connect()
+            if (connection.responseCode == HttpURLConnection.HTTP_OK) {
+                val inputStream = connection.inputStream
+                val fileName = url.substringAfterLast("/")
+                val file = File(context.cacheDir, fileName)
+
+                file.outputStream().use { outputStream ->
+                    inputStream.copyTo(outputStream)
+                }
+
+                // Pass file to your document handling logic
+                addDocument(file.inputStream(), fileName, Readers.DocumentType.PDF)
```

Comment:
Instead of assuming that the document will be a PDF, we can check the file extension from `fileName` and determine the document type.